aiida_firecrest/transport.py

```diff
@@ -135,16 +282,29 @@ def __init__(
         url: str,
         token_uri: str,
         client_id: str,
-        client_secret: str | Path,
+        client_secret: str,  # | Path,
+        # unfortunately we cannot store client_secret as a Path, because it is not JSON serializable
```

Comment:
```suggestion
        client_secret: str,
```
Thanks for the explanation. Even I would comment it out or maybe you can support it by converting it to string?
```python
if isinstance(client_secret, Path):
    client_secret = str(client_secret)
```

Reply:
Thanks @agoscinski , 
It's only `aiida-core` who instantiates `FirecrestTransport` , with inputs that are already stored in database. 
These inputs were taken from users when setting up the firecrest transport plugin through `click` interface.
The point is database backend of  `aiida-core` store the necessarily inputs to in a `json` file. So if this is being reached that means `client_secret` was already stored as a `str` . Since instantiating `FirecrestTransport` is something always occur in back-end, then  I think this is not be necessarily.